eZ/Publish/Core/Persistence/Legacy/Content/Type/Gateway/DoctrineDatabase.php

```diff
@@ -1157,8 +1268,9 @@ public function deleteGroupAssignmentsForType($typeId, $status)
      * @param int $typeId
      * @param int $sourceVersion
      * @param int $targetVersion
+     * @param \eZ\Publish\API\Repository\Values\ContentType\FieldDefinition[] $fieldDefinitions
      */
-    public function publishTypeAndFields($typeId, $sourceVersion, $targetVersion)
+    public function publishTypeAndFields($typeId, $sourceVersion, $targetVersion, array $fieldDefinitions = [])
```

Comment:
Missing PhpDoc, missing abstract Gateway method prototype update.
But the same comment as in case of `deleteFieldDefinitionsForType`